partiql-parser/build.gradle.kts

```diff
@@ -23,7 +23,35 @@ dependencies {
     api(project(":partiql-ast"))
     api(project(":partiql-types"))
     implementation(Deps.ionElement)
-    implementation(Deps.antlrRuntime)
+    shadow(Deps.antlrRuntime)
+}
+
+val relocations = mapOf(
+    "org.antlr" to "org.partiql.thirdparty.antlr"
+)
+
+tasks.shadowJar {
+    dependsOn(tasks.named("generateGrammarSource"))
+    configurations = listOf(project.configurations.shadow.get())
+    archiveClassifier.set("shaded")
+    for ((from, to) in relocations) {
+        relocate(from, to)
+    }
+}
+
+// Workaround for https://github.com/johnrengelman/shadow/issues/651
+components.withType(AdhocComponentWithVariants::class.java).forEach { c ->
+    c.withVariantsFromConfiguration(project.configurations.shadowRuntimeElements.get()) {
+        skip()
+    }
+}
+
+// `antlr` dependency configuration adds the ANTLR API configuration (and Maven `compile` dependency scope on publish)
```

Comment:
This is a known issue w/ the ANTLR gradle plugin. Workaround here removes the `antlr` Gradle `api` (equivalent to Maven `compile`) dependency.

Reply:
There are some build issues following this change. I'll take another look